prepare_release.sh

```diff
@@ -0,0 +1,155 @@
+        return 1
+    fi
+    MAJOR_VERSION=$(echo "$1" | sed -e 's/^\([0-9]*\)\.\([0-9]*\)\.\([0-9]*\)$/\1/')
+    MINOR_VERSION=$(echo "$1" | sed -e 's/^\([0-9]*\)\.\([0-9]*\)\.\([0-9]*\)$/\2/')
+    PATCH_VERSION=$(echo "$1" | sed -e 's/^\([0-9]*\)\.\([0-9]*\)\.\([0-9]*\)$/\3/')
+    return 0
+}
+
+set -e
+
+if [[ -z "$1" ]]; then
+    echo "[ERROR] No version number provided."
+    echo "[INFO] Usage: ./prepare_release.sh <VERSION_NUMBER>"
+    exit 1
+fi
+
+
+#############################
+#  VALIDATE VERSION NUMBER  #
+#############################
+
+VERSION="$1"
+if ! parseVersion "$VERSION"; then
+    echo "[ERROR] Illegal version number provided. Version number must match semver."
+    exit 1
+fi
+
+PROJECT_FILE="FirebaseAdmin/FirebaseAdmin/FirebaseAdmin.csproj"
+CUR_VERSION=$(grep "<Version>" ${PROJECT_FILE} | awk -F '>' '{print $2}' | awk -F '<' '{print $1}')
+if [ -z "$CUR_VERSION" ]; then
+    echo "[ERROR] Failed to find the current version. Check ${PROJECT_FILE} for version declaration."
+    exit 1
+fi
+if ! parseVersion "$CUR_VERSION"; then
+    echo "[ERROR] Illegal current version number. Version number must match semver."
+    exit 1
+fi
+
+if ! isNewerVersion "$VERSION" "$CUR_VERSION"; then
+    echo "[ERROR] Illegal version number provided. Version $VERSION <= $CUR_VERSION"
+    exit 1
+fi
+
+
+#############################
+#  VALIDATE TEST RESOURCES  #
+#############################
+
+INTEGRATION_TESTS_DIR="FirebaseAdmin/FirebaseAdmin.IntegrationTests/resources"
+if [[ ! -e "${INTEGRATION_TESTS_DIR}/integration_cert.json" ]]; then
+    echo "[ERROR] integration_cert.json file is required to run integration tests."
+    exit 1
+fi
+
+if [[ ! -e "${INTEGRATION_TESTS_DIR}/integration_apikey.txt" ]]; then
+    echo "[ERROR] integration_apikey.txt file is required to run integration tests."
+    exit 1
+fi
+
+
+###################
+#  VALIDATE REPO  #
+###################
+
+# Ensure the checked out branch is master
+CHECKED_OUT_BRANCH="$(git branch | grep "*" | awk -F ' ' '{print $2}')"
+if [[ $CHECKED_OUT_BRANCH != "master" ]]; then
+    read -p "[WARN] You are on the '${CHECKED_OUT_BRANCH}' branch, not 'master'. Continue? (Y/n) " CONTINUE
+    echo
+
+    if ! [[ $CONTINUE == "Y" ]]; then
+        echo "[INFO] You chose not to continue."
+        exit 1
+    fi
+fi
+
+# Ensure the branch does not have local changes
+if [[ $(git status --porcelain) ]]; then
+    read -p "[WARN] Local changes exist in the repo. Continue? (Y/n) " CONTINUE
+    echo
+
+    if ! [[ $CONTINUE == "Y" ]]; then
+        echo "[INFO] You chose not to continue."
+        exit 1
+    fi
+fi
+
+
+##################################
+#  UPDATE VERSION AND CHANGELOG  #
+##################################
+
+HOST=$(uname)
+echo "[INFO] Updating FirebaseAdmin.csproj and CHANGELOG.md"
+sed -i -e "s/<Version>$CUR_VERSION<\/Version>/<Version>$VERSION<\/Version>/" "${PROJECT_FILE}"
+sed -i -e "1 s/# Unreleased//" "CHANGELOG.md"
```

Comment:
Instead of having an in-place sed and then piping another sed into cat, just use awk:

    awk '
    BEGIN { print "# Unreleased\n\n-\n\n# v'${VERSION}'" }
    /^# Unreleased$/ { next }
    { print }
    ' CHANGELOG.md >TEMP_CHANGELOG.md

That also has the benefit of not adding an extra newline after the version.

Reply:
Done